### PR TITLE
align transformer tts

### PR DESCRIPTION
**Pull request by yt605155624**

### CODE REVIEW

File: examples/transformer_tts/ljspeech/conf/default.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    dprenet_dropout_rate: 0.5        # dropout rate for decoder prenet
+    postnet_dropout_rate: 0.5        # dropout rate for postnet
+    transformer_enc_dropout_rate: 0.1                # dropout rate for transformer encoder layer
+    transformer_enc_positional_dropout_rate: 0.1     # dropout rate for transformer encoder positional encoding
+    transformer_enc_attn_dropout_rate: 0.1           # dropout rate for transformer encoder attention layer
+    transformer_dec_dropout_rate: 0.1                # dropout rate for transformer decoder layer
+    transformer_dec_positional_dropout_rate: 0.1     # dropout rate for transformer decoder positional encoding
+    transformer_dec_attn_dropout_rate: 0.1           # dropout rate for transformer decoder attention layer
+    transformer_enc_dec_attn_dropout_rate: 0.1       # dropout rate for transformer encoder-decoder attention layer
+    num_heads_applied_guided_attn: 2                 # number of layers to apply guided attention loss
```
Comment (zh794390558) [inline examples/transformer_tts/ljspeech/conf/default.yaml]: 注释写反了吧？

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        self.msg = "Rank: {}, ".format(dist.get_rank())
+        losses_dict = {}
+
+        after_outs, before_outs, logits, ys, labels, olens, ilens, need_dict = self.model(
+            text=batch["text"],
+            text_lengths=batch["text_lengths"],
+            speech=batch["speech"],
+            speech_lengths=batch["speech_lengths"], )
+
+        self.criterion = TransformerTTSLoss(
```
Comment (zh794390558) [inline examples/transformer_tts/ljspeech/transformer_tts_updater.py]: 这部分应该可以放到init里，每次forward都要创建，费时。

File: parakeet/models/transformer_tts.py [hunk 3]
```diff
@@ -11,615 +11,1073 @@
...
+                self.num_layers_applied_guided_attn = num_layers_applied_guided_attn
+            if num_heads_applied_guided_attn == -1:
+                self.num_heads_applied_guided_attn = aheads
+            else:
+                self.num_heads_applied_guided_attn = num_heads_applied_guided_attn
+        if self.spk_embed_dim is not None:
+            self.spk_embed_integration_type = spk_embed_integration_type
+
+        # use idx 0 as padding idx
+        self.padding_idx = 0
```
Comment (zh794390558) [inline parakeet/models/transformer_tts.py:255]: 这个和vocab的预处理相关。

File: parakeet/modules/tacotron2/decoder.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            Dimension of the outputs.
+        n_layers : int, optional
+            The number of prenet layers.
+        n_units : int, optional
+            The number of prenet units.
+        """
+        super().__init__()
+        self.dropout_rate = dropout_rate
+        self.prenet = nn.LayerList()
+        self.dropout = nn.Dropout(self.dropout_rate)
```
Comment (zh794390558) [inline parakeet/modules/tacotron2/decoder.py]: del self.dropout

File: parakeet/modules/tacotron2/decoder.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tacotron2 decoder related modules."""
+
+import six
+import paddle.nn.functional as F
+from paddle import nn
+
+
+# postnet 应该放在这
```
Comment (zh794390558) [inline parakeet/modules/tacotron2/decoder.py]: del

File: examples/transformer_tts/ljspeech/conf/default.yaml (hunk 1 above)
Comment (yt605155624) [reply examples/transformer_tts/ljspeech/conf/default.yaml]: done

File: parakeet/modules/tacotron2/decoder.py (hunk 4 above)
Comment (yt605155624) [reply parakeet/modules/tacotron2/decoder.py]: done

File: parakeet/modules/tacotron2/decoder.py (hunk 5 above)
Comment (yt605155624) [reply parakeet/modules/tacotron2/decoder.py]: done

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py (hunk 2 above)
Comment (yt605155624) [reply examples/transformer_tts/ljspeech/transformer_tts_updater.py]: done

File: parakeet/models/transformer_tts.py (hunk 3 above)
Comment (yt605155624) [reply parakeet/models/transformer_tts.py:255]: 是的，预处理的时候会在token_list 的开头和末尾 <pad> 0 和 <eos> 76，fastspeech2 也是这么做的

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py [hunk 6]
```diff
@@ -215,10 +222,7 @@ def evaluate_core(self, batch):
             text_lengths=batch["text_lengths"],
             speech=batch["speech"],
             speech_lengths=batch["speech_lengths"])
-        self.criterion = TransformerTTSLoss(
-            use_masking=self.use_masking,
-            use_weighted_masking=self.use_weighted_masking,
-            bce_pos_weight=self.bce_pos_weight)
+
         if self.use_guided_attn_loss:
```
Comment (zh794390558) [inline examples/transformer_tts/ljspeech/transformer_tts_updater.py]: use_guided_attn_loss

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py [hunk 7]
```diff
@@ -70,10 +75,6 @@ def update_core(self, batch):
             speech=batch["speech"],
             speech_lengths=batch["speech_lengths"], )
 
-        self.criterion = TransformerTTSLoss(
-            use_masking=self.use_masking,
-            use_weighted_masking=self.use_weighted_masking,
-            bce_pos_weight=self.bce_pos_weight)
         if self.use_guided_attn_loss:
```
Comment (zh794390558) [inline examples/transformer_tts/ljspeech/transformer_tts_updater.py]: use_guided_attn_loss

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py (hunk 6 above)
Comment (yt605155624) [reply examples/transformer_tts/ljspeech/transformer_tts_updater.py]: done

File: examples/transformer_tts/ljspeech/transformer_tts_updater.py (hunk 7 above)
Comment (yt605155624) [reply examples/transformer_tts/ljspeech/transformer_tts_updater.py]: done

**Review by zh794390558 — APPROVED** [review]
LGTM
